packages/service/src/utils/documents.ts

```diff
@@ -0,0 +1,54 @@
+import * as Y from "yjs";
+import { db, pool, schema } from "../db";
+
+const TEXT_NAME = "codemirror";
+
+export function insertUpdate(document_id: string, document_update: Uint8Array) {
+  return db
+    .insert("document_updates_queue", {
+      document_id,
+      document_update: Buffer.from(document_update),
+    })
+    .run(pool);
+}
+
+export const createDocument = (doc: Y.Doc) =>
+  db
+    .insert("document", {
+      value: doc.getText(TEXT_NAME).toJSON(),
+      origin: Buffer.from(Y.encodeStateAsUpdate(doc)),
+      web_rtc_key: "",
+      latest_update_time: new Date(),
+    })
+    .run(pool);
+
+type DocumentWithUpdates = schema.document.Selectable & {
+  document_updates?: schema.document_updates_queue.Selectable[];
+};
+
+export function fetchDocument(id: string): Promise<DocumentWithUpdates> {
+  return db
+    .selectExactlyOne(
+      "document",
+      { id },
+      {
+        lateral: {
+          document_updates: db.select("document_updates_queue", {
+            document_id: db.parent("id"),
+            created_at: db.conditions.gt(db.parent("latest_update_time")),
+          }),
+        },
+      }
+    )
+    .run(pool)
+    .then((doc: any) => ({
```

Comment:
I think the types of these lateral joins that zapatos is resolving are wrong. They didn't include the document_updates array that I wanted. Also, it isn't handling data conversions for me for some reason (Buffers need to be `Buffers`, and Dates need to be `Dates`).

Reply:
Was the raw sql query you had before working as expected?